src/EXTRA-FIX/fix_wall_flow.cpp

```diff
@@ -0,0 +1,294 @@
+
+  ++iarg;
+  // parsing temperature
+  double flowtemp = utils::numeric(FLERR,arg[iarg],do_abort,lmp);
+  kT = lmp->force->boltz * flowtemp / force->mvv2e;
+
+  ++iarg;
+  // parsing seed
+  rndseed = utils::inumeric(FLERR, arg[iarg],do_abort,lmp);
+  if(rndseed <= 0) error->all(FLERR, "Illegal fix wall/flow argument: random seed must be positive integer");
+
+  ++iarg;
+  // parsing wall count
+  int wallcount = utils::inumeric(FLERR,arg[iarg],do_abort,lmp);
+  if(wallcount <= 0) error->all(FLERR,"Illegal fix wall/flow argument: wall count must be positive integer");
+
+  ++iarg;
+  // parsing walls
+  if(narg - iarg != wallcount && narg - iarg != wallcount + 2) error->all(FLERR, "Wrong fix wall/flow wall count");
+  auto getscale = [&]() -> double {
+      switch (flowax)
+      {
+        case FlowAxis::AX_X:
+          return domain->lattice->xlattice;
+        case FlowAxis::AX_Y:
+          return domain->lattice->ylattice;
+        case FlowAxis::AX_Z:
+          return domain->lattice->zlattice;
+        default: return 0.0;
+      }
+      return 0.0;
+  };
+  double scale = getscale();
+
+  if (narg - iarg == wallcount + 2)
+  {
+    if(strcmp(arg[narg - 2], "units") != 0) error->all(FLERR, "Wrong fix wall/flow units command");
+    if (strcmp(arg[narg - 1], "box") == 0) scale = 1.0;
+    else if (strcmp(arg[narg - 1], "lattice") == 0)
+    {
+      scale = getscale();
+    }
+    else error->all(FLERR, "Wrong fix wall/flow units command");
+  }
+
+  walls.resize(wallcount + 2);
+  walls.front() = domain->boxlo[flowax];
+  for (size_t w = 1; w <= wallcount; ++w, ++iarg)
+  {
+    walls[w] = utils::numeric(FLERR,arg[iarg],do_abort,lmp) * scale;
+  }
+  walls.back() = domain->boxhi[flowax];
+  if (!std::is_sorted(walls.begin(), walls.end(), std::less_equal<double>()))
+  {
+    error->all(FLERR, "Wrong fix wall/flow wall ordering or some walls are outside simulation domain");
+  }
+  std::cout << "Walls:\n";
+  for (auto w : walls)
+  {
+    std::cout << w << " ";
+  }
+  std::cout << std::endl;
+
+  memory->grow(current_segment, atom->nmax, "WallFlow::current_segment");
+  atom->add_callback(Atom::GROW);
+  if (restart_peratom) atom->add_callback(Atom::RESTART);
+
+  maxexchange = 1;
+
+  random = new RanMars(lmp, rndseed + comm->me);
+ }
+
+/* ---------------------------------------------------------------------- */
+
+FixWallFlow::~FixWallFlow()
+{
+  if (copymode) return;
+  atom->delete_callback(id, Atom::GROW);
+  if (restart_peratom) atom->delete_callback(id, Atom::RESTART);
+  memory->destroy(current_segment);
+
+  delete random;
+}
+
+/* ---------------------------------------------------------------------- */
+
+int FixWallFlow::setmask()
+{
+  int mask = 0;
+
+  mask |= END_OF_STEP;
+
+  return mask;
+}
+
+/* ---------------------------------------------------------------------- */
+
+void FixWallFlow::init()
+{
+  int nrigid = 0;
+  for (int i = 0; i < modify->nfix; i++)
```

Comment:
This kind of construct is strongly discouraged since it requires accessing internal data of the Modify class.
Instead you can use:
```
for (auto ifix : modify->get_fix_list()) if (ifix->rigid_flag) nrigid++;
```